sockeye/training.py

```diff
@@ -188,6 +296,14 @@ def create_eval_metric_composite(metric_names: List[AnyStr]) -> mx.metric.Compos
         metrics = [TrainingModel.create_eval_metric(metric_name) for metric_name in metric_names]
         return mx.metric.create(metrics)
 
+    def _set_states(self, state_names, values):
+        states = self.module.get_states(merge_multi_context=False)
+        for state_name, state, value in zip(state_names, states, values):
+            # ACCLIFT: this is something that existed in the previous initial implementation.
```

Comment:
This seems related to the scheduling where you need the # of updates symbolically in the graph.